### ISSUE TITLE
codepipeline_actions: No way to add CodeConnection source

### ISSUE THREAD

**Issue by revmischa** [aws/aws-cdk#30957]
### Describe the feature

I'm trying to set up CodePipeline with a GitHub source. I think this is a common flow.
I see that there is a `GitHubSourceAction` construct but it's not what I want - it wants an oauth secret. But now in the console we can go set up a "CodeConnection" to link GItHub to AWS without having to store an oauth token somewhere in secrets manager. I'd like to use that existing connection someone has created.

I found the L1 construct I want, but have no way to use it as a Source Action in CodePipeline: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_codeconnections.CfnConnection.html

Please add support for a CodeConnections Source Action. 

### Use Case

I think my use case is pretty common: using GitHub with CodePipeline.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.150.0

### Environment details (OS name and version, etc.)

Linux

**Comment by khushail** [comment]
@revmischa , yes, having CodeConnection [source](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines.CodePipelineSource.html) would be useful as we already have [CodeStarConnection](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines.ConnectionSourceOptions.html) as source in CDK. Thanks for requesting this!

**Comment by ishon19** [comment]
Hi @khushail, I would like to work on this issue!

**Comment by khushail** [comment]
Hey @ishon19 , thanks for volunteering for the contribution of PR. Here is the [contribution guide](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-1-find-something-to-work-on) to get started on the process.
Once your PR is ready, you could reach out to [cdk.dev](https://cdk.dev/) community for review.

**Comment by ishon19** [comment]
> Hey @ishon19 , thanks for volunteering for the contribution of PR. Here is the [contribution guide](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-1-find-something-to-work-on) to get started on the process. Once your PR is ready, you could reach out to [cdk.dev](https://cdk.dev/) community for review.

Absolutely, thanks @khushail!

**Comment by chetancyberoi** [comment]
Any updates, I am basically stuck here

**Comment by zytalus** [comment]
How to use Code Connection with AWS CDK CI/CD Pipeline.

Create a AWS Code Connection by following steps here https://docs.aws.amazon.com/dtconsole/latest/userguide/connections-create-github.html

Replace CodePipelineSource.gitHub with CodePipelineSource.connection

Example*
```
    const pipeline = new CodePipeline(this, 'MyPipeline', {
      pipelineName: 'MyPipeline',
      synth: new ShellStep('Synth', {
        input: CodePipelineSource.connection('OWNER/REPO', 'main', {
          connectionArn: 'arn:aws:codestar-connections:us-east-1:123456789012:connection/12345678-abcd-12ab-34cdef5678gh'
        }),
        commands: ['npm ci', 'npm run build', 'npx cdk synth']
      })
    });
```

**Comment by eggplants** [comment]
Can I use `CodeStarConnectionsSourceAction` for new AWS CodeConnections resource instead of absent `CodeConnectionSourceAction`?

**Comment by danielsharvey** [comment]
This worked for me: https://github.com/aws/aws-cdk/issues/32443#issuecomment-2528454972
